.github/workflows/update-kaveri-data.yml

```diff
@@ -0,0 +1,157 @@
+name: Automated Kaveri Data Update
+
+on:
+  schedule:
+    # Run weekly full scrape on Sunday at 3 AM UTC  
+    - cron: '0 3 * * 0'
+  
+  workflow_dispatch:
+    inputs:
+      force_full_scrape:
+        description: 'Force full scrape regardless of changes'
+        required: false
+        default: false
+        type: boolean
+
+permissions:
+  contents: write
+  actions: read
+  issues: write
+
+jobs:
+  update-kaveri-data:
+    runs-on: ubuntu-latest
+    
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v4
+      with:
+        token: ${{ secrets.GITHUB_TOKEN }}
+    
+    - name: Set up Python
+      uses: actions/setup-python@v4
+      with:
+        python-version: '3.9'
+    
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
```

Comment:
Vendor dependencies for reproducibility. Poetry would be good for this with lockfile support.